src/core/net/dnssd_server.hpp

```diff
@@ -232,46 +236,111 @@ class Server : public InstanceLocator, private NonCopyable
+                                            const char *      aHostName,
+                                            uint32_t          aTtl,
+                                            uint16_t          aPriority,
+                                            uint16_t          aWeight,
+                                            uint16_t          aPort,
+                                            NameCompressInfo &aCompressInfo);
+    static otError          AppendTxtRecord(Message &         aMessage,
+                                            const char *      aInstanceName,
+                                            const void *      aTxtData,
+                                            uint16_t          aTxtLength,
+                                            uint32_t          aTtl,
+                                            NameCompressInfo &aCompressInfo);
+    static otError          AppendAaaaRecord(Message &           aMessage,
+                                             const char *        aHostName,
+                                             const Ip6::Address &aAddress,
+                                             uint32_t            aTtl,
+                                             NameCompressInfo &  aCompressInfo);
+    static otError          AppendServiceName(Message &aMessage, const char *aName, NameCompressInfo &aCompressInfo);
+    static otError          AppendInstanceName(Message &aMessage, const char *aName, NameCompressInfo &aCompressInfo);
+    static otError          AppendHostName(Message &aMessage, const char *aName, NameCompressInfo &aCompressInfo);
+    static void             IncResourceRecordCount(Header &aHeader, bool aAdditional);
+    static otError          FindNameComponents(const char *aName, const char *aDomain, NameComponentsOffsetInfo &aInfo);
+    static otError          FindPreviousLabel(const char *aName, uint8_t &aStart, uint8_t &aStop);
+    static void             SendResponse(Header                  aHeader,
+                                         Header::Response        aResponse,
+                                         Message &               aMessage,
+                                         const Ip6::MessageInfo &aMessageInfo,
+                                         Ip6::Udp::Socket &      aSocket);
 
 #if OPENTHREAD_CONFIG_SRP_SERVER_ENABLE
+    Header::Response                   ResolveBySrp(const Header &            aRequestHeader,
+                                                    const Message &           aRequestMessage,
+                                                    Header &                  aResponseHeader,
+                                                    Message &                 aResponse,
+                                                    uint8_t                   aResolveAdditional,
+                                                    Server::NameCompressInfo &aCompressInfo);
```

Comment:
```suggestion
    Header::Response                   ResolveBySrp(DnsQueryTransaction &aDnsQuery);
```

Reply:
Let's postpone this major refactoring and deliver the working feature first.